Sources/VGSCheckoutSDK/Core/Collect/CollectUIElements/Text Field/PickerFields/AddressDataProviders/VGSAddressDataProvider.swift

```diff
@@ -37,11 +37,45 @@ internal class VGSAddressCountriesDataProvider {
 		}
 	}
 
+  /// List of  country models that match provided valid `countryISOCodes`. Order will be the same as order in `countryISOCodes`. Returns all countries if no valid `countryISOCodes`.
+  static func provideCountriesWithISOCode(_ countryISOCodes: [String]?) -> [CountryModel] {
+    let allCountries = provideAllCountries()
+    guard let countryCodes = countryISOCodes, !countryCodes.isEmpty else {
+      let message = "No valid country ISO Codes provided. All countries will be used."
+      let event = VGSLogEvent(level: .warning, text: message, severityLevel: .error)
+      VGSCheckoutLogger.shared.forwardLogEvent(event)
+      return allCountries
+    }
+    
+    var validCountryModels = [CountryModel]()
+    var invalidCountryISOCodes = [String]()
+    for countryCode in countryCodes {
+      if let countryModel = allCountries.first(where: {$0.code == countryCode}) {
+        validCountryModels.append(countryModel)
+      } else {
+        invalidCountryISOCodes.append(countryCode)
+      }
+    }
```

Comment:
This code works well would also suggest to cover next cases:
1. try to `uppercase` each string in array so ["ca"] will be transformed to ["CA"];
2. remove possible duplications, current code doesn't handle this case and it might break UI.

`checkoutConfiguration.billingAddressCountryFieldOptions.validCountries = ["AU", "AU", "AU"] `
 
 Australia will be displayed 3 times.